src/test/scala/chiselTests/util/BitPatSpec.scala

```diff
@@ -48,4 +49,32 @@ class BitPatSpec extends AnyFlatSpec with Matchers {
     b(4, 3) should be(BitPat("b01"))
     b(6, 6) should be(BitPat("b1"))
   }
+
+  it should "construct BitPat from Bundle" in {
+    import chisel3.experimental.ChiselEnum
+
+    object MyEnum extends ChiselEnum {
+      val sA, sB = Value
+    }
+
+    class MyBundle extends Bundle {
+      class YourBundle extends Bundle {
+        val b = Bool()
+        val c = MyEnum()
+      }
+      val a = UInt(8.W)
+      val b = new YourBundle
+      val c = MyEnum()
+    }
```

Comment:
```suggestion
    object MyBundle {
      class YourBundle extends Bundle {
        val b = Bool()
        val c = MyEnum()
      }
      object YourBundle {
        implicit val ybBP: GenBitPat[YourBundle] = genBitPat {
          b -> BitPat(true.B)
          c -> ...
        }
      }
    }
    class MyBundle extends Bundle {

      val a = UInt(8.W)
      val b = new YourBundle
      val c = MyEnum()
    }
    
    (new MyBundle).bp { b =>
       Map(
         b.a -> BitPat(5.U(8.W)), // 00000101
         b.c -> BitPat(MyEnum.sB.litValue.U(MyEnum.getWidth.W)), // 1
         b.b.c -> b.b.c.bp()
       )
     } == BitPat("b1_0_?_0_0000101")    
```
Sketch of a typeclass pattern for Bundle BitPats